lib/http_aws_sigv4.c

```diff
@@ -427,13 +470,13 @@ CURLcode Curl_output_aws_sigv4(struct Curl_easy *data, bool proxy)
                     "%s\n" /* CanonicalQueryString */
                     "%s\n" /* CanonicalHeaders */
                     "%s\n" /* SignedHeaders */
-                    "%s",  /* HashedRequestPayload in hex */
+                    "%.*s",  /* HashedRequestPayload in hex */
                     method,
                     data->state.up.path,
                     data->state.up.query ? data->state.up.query : "",
                     Curl_dyn_ptr(&canonical_headers),
                     Curl_dyn_ptr(&signed_headers),
-                    sha_hex);
+                    payload_hash_len, payload_hash);
```

Comment:
the length should be an `int` according to how *printf() works, but I think using the `%.*s` construct seems unnecessary here. Can't you just rather do `%s` and `payload_hash_len ? payload_hash: ""` as argument?

Reply:
> Can't you just rather do %s and payload_hash_len ? payload_hash: "" as argument?

`parse_content_sha_hdr()` trims any trailing whitespace from the header value, so may return a shorter length

`payload_hash` points to memory returned by `Curl_checkheaders()`, so i used the explicit `payload_hash_len` here to avoid mutating that header to move the null terminator

i considered copying this header value into a different buffer to change the null. but that required the choice of a buffer size, so added an extra error path when the user provides a header value that's longer

i found this solution to be the simplest, but would welcome your guidance here

> the length should be an int according to how *printf() works

ok, thanks